upup/pkg/fi/executor.go

```diff
@@ -171,39 +173,85 @@ func (e *executor[T]) RunTasks(taskMap map[string]Task[T]) error {
 	return nil
 }
 
+// getExecutorPoolSize determines the executor pool size for executing tasks in parallel
+//
+// It will determine the pool size based on the current number of tasks and the maximum pool size.
+// If the maximum pool size is not specified or greater than the actual number of tasks, then the function returns the number of tasks as the pool size.
+// Otherwise it will return the defined maximum pool size.
+//
+// Currently, the max pool size can be set by specifying the ENV variable "KOPS_EXECUTOR_POOL_MAX_SIZE".
+func (e *executor[T]) getExecutorPoolSize(numberOfTasks int) int {
+	// TODO: make configurable via flag rather than env var?
+	executorPoolMaxSize, _ := strconv.Atoi(os.Getenv("KOPS_EXECUTOR_POOL_MAX_SIZE"))
```

Comment:
Agreeing this value is probably the hardest thing here.  TBH I'd probably hard-code it to 4 or some similar value that you have found works well (initially), just because we need to exercise the code to find any bugs.

We could also see if there's some way to just set it higher in all the e2e tests, but I expect that we are generally bottlenecked on waiting on cloud APIs, so I think we can justify some concurrency by default.  If we find that some cloud is always throttling us we can set the default to 1 on that cloud.